File: backend/models/audit_log.py
from pydantic import BaseModel, Field
from typing import Optional
from datetime import datetime


class AuditLogOut(BaseModel):
    id: str = Field(alias="_id")
    user: Optional[dict] = None
    userName: str
    action: str
    resource: str
    resourceId: str = ""
    details: str = ""
    ipAddress: str = ""
    userAgent: str = ""
    createdAt: Optional[datetime] = None
    updatedAt: Optional[datetime] = None

    class Config:
        populate_by_name = True

File: frontend/models/AuditLog.js
import mongoose from 'mongoose'

const AuditLogSchema = new mongoose.Schema({
  user: { type: mongoose.Schema.Types.ObjectId, ref: 'User' },
  userName: { type: String, required: true },
  action: { type: String, required: true },
  resource: { type: String, required: true },
  resourceId: { type: String, default: '' },
  details: { type: String, default: '' },
  ipAddress: { type: String, default: '' },
  userAgent: { type: String, default: '' }
}, {
  timestamps: true,
  collection: 'auditlogs'
})

export default mongoose.models.AuditLog || mongoose.model('AuditLog', AuditLogSchema)
